File: mo-system/mo-system-service/src/main/java/ink/ckx/mo/system/service/SysMenuService.java
package ink.ckx.mo.system.service;

import com.baomidou.mybatisplus.extension.service.IService;
import ink.ckx.mo.common.web.model.Option;
import ink.ckx.mo.system.api.model.entity.SysMenu;
import ink.ckx.mo.system.api.model.form.MenuForm;
import ink.ckx.mo.system.api.model.query.MenuListQuery;
import ink.ckx.mo.system.api.model.vo.menu.MenuDetailVO;
import ink.ckx.mo.system.api.model.vo.menu.MenuListVO;
import ink.ckx.mo.system.api.model.vo.menu.ResourceListVO;
import ink.ckx.mo.system.api.model.vo.menu.RouteListVO;

import java.util.List;

/**
 * 菜单业务接口
 *
 * @author chenkaixin
 */
public interface SysMenuService extends IService<SysMenu> {

    /**
     * 获取菜单表格列表
     *
     * @return
     */
    List<MenuListVO> listMenus(MenuListQuery menuListQuery);

    /**
     * 获取菜单下拉列表
     *
     * @return
     */
    List<Option<Long>> listMenuOptions();

    /**
     * 新增菜单
     *
     * @param menuForm
     * @return
     */
    Long saveMenu(MenuForm menuForm);

    /**
     * 更新菜单
     *
     * @param menuId
     * @param menuForm
     * @return
     */
    void updateMenu(Long menuId, MenuForm menuForm);

    /**
     * 删除菜单
     *
     * @param menuId
     * @return
     */
    void deleteMenu(Long menuId);

    /**
     * 获取路由列表
     *
     * @return
     */
    List<RouteListVO> listRoutes();

    /**
     * 获取菜单详情
     *
     * @return
     */
    MenuDetailVO getMenuDetail(Long id);

    /**
     * 资源树形列表
     *
     * @return
     */
    List<ResourceListVO> listResources();

    /**
     * 修改菜单显示状态
     *
     * @param menuId  菜单ID
     * @param visible 是否显示
     * @return
     */
    void updateMenuVisible(Long menuId, Integer visible);

    /**
     * 刷新Redis缓存中角色权限
     *
     * @return
     */
    void refreshRolePerm();
}

File: mo-admin/mo-admin-service/src/main/kotlin/ink/ckx/mo/admin/service/SysMenuService.kt
package ink.ckx.mo.admin.service

import com.baomidou.mybatisplus.extension.service.IService
import ink.ckx.mo.admin.api.model.entity.SysMenu
import ink.ckx.mo.admin.api.model.form.MenuForm
import ink.ckx.mo.admin.api.model.query.MenuListQuery
import ink.ckx.mo.admin.api.model.vo.menu.MenuDetailVO
import ink.ckx.mo.admin.api.model.vo.menu.MenuListVO
import ink.ckx.mo.admin.api.model.vo.menu.ResourceListVO
import ink.ckx.mo.admin.api.model.vo.menu.RouteListVO
import ink.ckx.mo.common.web.enums.StatusEnum
import ink.ckx.mo.common.web.model.Option

/**
 * 菜单业务接口
 *
 * @author chenkaixin
 */
interface SysMenuService : IService<SysMenu> {

    /**
     * 获取菜单表格列表
     *
     * @return
     */
    fun listMenus(menuListQuery: MenuListQuery): List<MenuListVO>

    /**
     * 获取菜单下拉列表
     *
     * @return
     */
    fun listMenuOptions(): List<Option<Any>>

    /**
     * 新增菜单
     *
     * @param menuForm
     * @return
     */
    fun saveMenu(menuForm: MenuForm): Long?

    /**
     * 更新菜单
     *
     * @param menuId
     * @param menuForm
     * @return
     */
    fun updateMenu(menuId: Long, menuForm: MenuForm)

    /**
     * 删除菜单
     *
     * @param menuId
     * @return
     */
    fun deleteMenu(menuId: Long)

    /**
     * 获取路由列表
     *
     * @return
     */
    fun listRoutes(): List<RouteListVO>

    /**
     * 获取菜单详情
     *
     * @return
     */
    fun getMenuDetail(menuId: Long): MenuDetailVO

    /**
     * 资源树形列表
     *
     * @return
     */
    fun listResources(): List<ResourceListVO>

    /**
     * 修改菜单显示状态
     *
     * @param menuId  菜单ID
     * @param visible 是否显示
     * @return
     */
    fun updateMenuVisible(menuId: Long, visible: StatusEnum)

    /**
     * 刷新Redis缓存中角色权限
     *
     * @return
     */
    fun refreshRolePerm()
}